modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobContainer.java

```diff
@@ -300,6 +304,11 @@ long getLargeBlobThresholdInBytes() {
         return blobStore.bufferSizeInBytes();
     }
 
+    // package private for testing
+    long getLargeCopyThresholdInBytes() {
+        return MAX_FILE_SIZE.getBytes();
+    }
```

Comment:
Could we make this a setting similar to the `buffer_size` and `chunk_size` ones? Ideally imposing the same limitations as S3 proper (so a minimum of 5MiB and a max of 5GiB)? That way we can adjust it in `S3RepositoryAnalysisRestIT` to get the coverage of both paths there.

Reply:
Done in https://github.com/elastic/elasticsearch/pull/125737/commits/a9ff4b926ffd68d060271fbb6967b2d108b7edab